TrackletAlgorithm/TrackletEngineUnit.h

```diff
@@ -15,8 +15,8 @@ class TrackletEngineUnit {
     kNBitsRZFine=3,
     kNBitsPhiBins=3,
     kNBitsNegDiskSize=1,
-    kNBitsPTLutInner=(Seed==TF::L5L6||TF::L1D1||TF::L2D1||TF::D1D2||TF::D3D4)?1024:(Seed==(TF::L1L2||TF::L2L3)?256:512),
-    kNBitsPTLutOuter=(Seed==TF::L5L6||TF::L1D1||TF::L2D1||TF::D1D2||TF::D3D4)?1024:(Seed==(TF::L1L2||TF::L2L3||TF::L3L4)?256:512)
+    kNBitsPTLutInner=(Seed==TF::L5L6||TF::L1D1||TF::L2D1||TF::D1D2||TF::D3D4)?1024:(Seed==(TF::L1L2||TF::L2L3||TF::L3L4)?256:512),
+    kNBitsPTLutOuter=(Seed==TF::L5L6||TF::L1D1||TF::L2D1||TF::D1D2||TF::D3D4)?1024:(Seed==(TF::L1L2||TF::L2L3)?256:512)
```

Comment:
@jasonfan393 This doesn't make any sense, even before this PR. I think what these lines are trying to do is what I now have in #304 here (another argument in favor of adding the CXXFLAGS in that PR):
https://github.com/cms-L1TK/firmware-hls/blob/e785bf0b8565e4ae0a2a907a0cf1842cc3f9c620/TrackletAlgorithm/TrackletEngineUnit.h#L18-L19

The equality test has to be done for each seed value, instead of just logically ANDing the values together.

Reply:
@jasonfan393 P.S.: It's hard to imagine these lines are currently doing the right thing in all cases, so it's probably worth rechecking the resource utilization and timing after fixing them.

Reply:
yeah I'll fix, I had no idea how this was not causing simulation failures before when I looked at it

Reply:
Looks like this is probably the root cause of the weird LUT numbers - I'm testing all modules right now, but I will probably remove the L3L4 hack if all is well(im still not sure why this worked at all)